### PR TITLE
✨ Implement ready status packet handling system

### LINKED ISSUE #135: ✨ [Server] Implement READY_STATUS packet handling system
## Is your feature request related to a problem? Please describe

Currently, the server accepts TCP connections and handles `CONNECT_REQ` packets, but there is no mechanism for clients to signal their "ready" status in a lobby before game start. The `ClientConnection` struct has a `ready_` field that is initialized to `false` but never updated. According to the protocol specification (OpCode `0x07` READY_STATUS), clients should be able to send their ready state, and the server should broadcast this to all connected players and trigger game start when all players are ready.

## Describe the solution you'd like

Implement a complete server-side ready system that:

1. **Handles `READY_STATUS` packets (OpCode `0x07`):**
   - Listen for incoming TCP packets with OpCode `0x07` from connected clients
   - Deserialize the packet to extract `PlayerId` and `ReadyState` (boolean)
   - Validate that the `PlayerId` matches an existing connected client
   - Update the corresponding `ClientConnection::ready_` field

2. **Broadcasts ready state changes:**
   - When a player's ready state changes, send a `READY_STATUS` packet to all other connected clients
   - This allows clients to display which players are ready in the lobby UI

3. **Triggers game start when all players are ready:**
   - Check if all connected clients have `ready_ == true`
   - If so, broadcast a `GAME_START` packet (OpCode `0x06`) to all clients
   - Transition server state to "in-game" mode (spawn initial entities, start game tick)

4. **Follows the protocol specification:**
   - Packet structure as defined in protocol.md and network-rfc.txt
   - TCP transport for reliable delivery
   - Proper error handling (malformed packets, invalid player IDs, disconnect during ready phase)

5. **Implementation details:**
   - Add `HandleReadyStatus()` method in `Server.cpp` similar to `HandleConnectReq()`
   - Modify TCP receive logic to route OpCode `0x07` to the ready handler
   - Add helper methods: `BroadcastReadyStatus()`, `CheckAllPlayersReady()`, `StartGame()`
   - Update `Server::Update()` to handle game state transitions (lobby → in-game)

## Describe alternatives you've considered

1. **Auto-ready on connect:** Players are automatically marked as ready when they connect. This simplifies implementation but removes player control over game start timing.

2. **HTTP REST API for lobby management:** Use a separate HTTP endpoint for lobby state instead of binary TCP packets. This adds complexity and doesn't align with the existing protocol design.

3. **UDP-based ready signaling:** Send ready status via UDP instead of TCP. This is unreliable and inappropriate for critical session management.

## Additional context

- **Related files:**
  - Server.cpp (add ready packet handler)
  - Server.hpp (add method declarations)
  - Network.hpp (TCP receive logic)
  - Packets.hpp (ensure `ReadyStatusPacket` exists)
  
- **Protocol reference:**
  - See protocol.md Section 4, OpCode `0x07` READY_STATUS
  - See network-rfc.txt Section 5.1 for TCP command specifications

- **Testing strategy:**
  - Unit test: `test_ready_packet_serialization.cpp` (verify packet format)
  - Integration test: Connect multiple mock clients, send ready packets, verify game start trigger
  - Edge case: Disconnect a client after marking ready (should update ready count)

- **Dependencies:**
  - Requires `ReadyStatusPacket` and `GameStartPacket` classes in Packets.hpp
  - Requires C++23, Asio (async TCP I/O), existing `ClientConnection` struct

- **Acceptance criteria:**
  - [ ] Server correctly deserializes `READY_STATUS` packets
  - [ ] `ClientConnection::ready_` field updates when packet received
  - [ ] Ready state changes are broadcast to all clients
  - [ ] Game starts (sends `GAME_START`) when all players are ready
  - [ ] Server handles disconnect during ready phase gracefully
  - [ ] All changes follow Google C++ Style Guide (UpperCamelCase methods, snake_case_ members)
  - [ ] Code includes Doxygen documentation for all new methods

### PR DESCRIPTION
**Pull request by Jocelyn-JE**
## 🚀 Pull Request Summary

Closes #135
Milestone: Milestone 3

---

## 🧩 Type of Change

Select all relevant options:

- [X] ✨ Feature
- [ ] 🐛 Bug Fix
- [X] ♻️ Refactor
- [ ] 📝 Documentation
- [ ] 🧪 Tests
- [ ] 🔧 Tooling / CI
- [ ] 🚀 Performance
- [ ] Other (please describe)

---

## 📦 Changes Included

List the main changes made in this PR:

- Refactored the server architecture for handling and sending packets
- Added ready functionality as well as autostart system

---

## 🧪 How to Test

Describe how reviewers can test your changes:

1. Build the project (`cmake --build build`)
2. Run the server
3. Use the `scripts/dummy_client.py` python script to test out connecting to the server
4. See server logs to confirm correct functioning

---

## 🗂️ Related Areas

Which part of the project does this PR affect?

- [ ] area/docs
- [X] area/server
- [ ] area/client
- [ ] area/engine
- [X] area/tests

---

## 🔍 Checklist Before Requesting Review

Please confirm the following items are done:

- [X] My commits follow the **Gitmoji + English commit message** convention
- [X] My branch name follows the required format (`feature/XX-name`)
- [X] I have linked the PR to an Issue (`Closes #XX`)
- [ ] I ran and passed **all Git hooks**
- [ ] I added or updated documentation if needed
- [ ] I added or updated tests if needed
- [X] The code builds without errors
- [X] I performed manual testing
- [X] CI passes all checks (build + tests + formatting)

### CODE REVIEW

**Review by Jocelyn-JE — COMMENTED** [review]
Reviewed everything I made just to be sure!

File: server/include/server/PacketSender.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        network::ConnectAckPacket::Status status,
+        uint8_t assigned_player_id = 0);
+
+    /**
+     * @brief Send GAME_START packet to all authenticated players
+     *
+     * RFC Section 5.5: Notifies all players that the game is starting.
+     * Only sends to clients with player_id != 0 (authenticated).
+     */
+    void SendGameStart();
```
[the remark this answers is not in the document]
Comment (Jocelyn-JE) [reply server/include/server/PacketSender.hpp]: ???
